src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,54 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static int AGE_AT_LEAST_REQUIRE = 18;
+    private static int LENGTH_AT_LEAST_REQUIRE = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNull(user);
+        checkAge(user.getAge());
+        checkLogin(user.getLogin());
+        checkPassword(user.getPassword());
+        storageDao.add(user);
+        return user;
```

Comment:
The method 'register' should return the result of 'storageDao.add(user)', not just the user object. This ensures that the user returned is the one that was actually added to the storage.